lib/mixlib/install/backend/artifactory.rb

```diff
@@ -54,24 +56,31 @@ def info
           end
         end
 
-        private
+        def product_versions
+          params = {
+            g: "com.getchef",
+            a: options.product_name,
+            repos: ARTIFACTORY_REPOSITORY
+          }
+          results = client.get("/api/search/versions", params)["results"]
```

Comment:
Ok cool. Note that I am not using Artifactory's latest version fetching. It's fetching all versions for a channel/product.